typescript/token/package.json

```diff
@@ -0,0 +1,62 @@
+{
+  "name": "@hyperlane-xyz/hyperlane-token",
```

Comment:
Wondering if this should be renamed to `@hyperlane-xyz/token` which is less redundant and matches other package names. Would mean we need to delete and republish the npm package but that's nbd. cc @yorhodes